kratos/processes/find_intersected_geometrical_objects_process.cpp

```diff
@@ -347,11 +347,20 @@ bool FindIntersectedGeometricalObjectsProcess::HasIntersection(
     GeometryType& rSecondGeometry
     )
 {
-    const IndexType work_dim = rFirstGeometry.WorkingSpaceDimension(); // TODO: DOMAIN_SIZE should be considered for consistency with other implementations
-    if (work_dim == 2) {
-        return this->HasIntersection2D(rFirstGeometry, rSecondGeometry);
+    const IndexType working_space_dimension = rFirstGeometry.WorkingSpaceDimension(); // TODO: DOMAIN_SIZE should be considered for consistency with other implementations
+    const IndexType local_space_dimension = rFirstGeometry.LocalSpaceDimension();
+    if (working_space_dimension == 2) {
+        if (local_space_dimension == 2) {
+            return this->HasIntersection2D(rFirstGeometry, rSecondGeometry);
+        } else {
+            return this->HasDirectIntersection2D(rFirstGeometry, rSecondGeometry);
```

Comment:
The difference of this new method with respect to the current `HasIntersection2D` is that it doesn't check that the intersecting geometry is inside the intersected candidate one. Am I wrong?

If this is the case, can you describe a use case in which it is required?

Reply:
As @rubenzorrilla asked, what is the difference?